xblock/mixins.py

```diff
@@ -522,7 +522,9 @@ def _add_field(self, node, field_name, field):
             elem.text = value
             node.insert(0, elem)
         else:
-            node.set(field_name, value)
+            # setting the None in the value for node raises TypeError to avoid this
```

Comment:
Yes, I would override `to_json` and `from_json` in the `String` field to get sensible behavior that accounts for None.